rootfs/api/models/app.py

```diff
@@ -393,13 +393,18 @@ def _default_structure(self, release):
         if not release.build.sha:
             structure = {'cmd': 1}
 
-        # if a dockerfile exists without a procfile, assume docker workflow
-        elif release.build.dockerfile and not release.build.procfile:
+        elif 'web' in release.build.procfile:
```

Comment:
this will fail if `release.build.procfile` == `None`. It'd be better to do

```
elif release.build.procfile and 'web' in release.build.procfile:
```

So we ensure that there's a procfile to parse before checking if there's a web process in there